File: server/src/main/java/com/smartexpense/api/application/port/in/ProfileUseCase.java
package com.smartexpense.api.application.port.in;

import com.smartexpense.api.application.dto.request.UpdateProfileRequestDto;
import com.smartexpense.api.application.dto.response.UserProfileResponseDto;

public interface ProfileUseCase {
    UserProfileResponseDto getProfile(String email);
    UserProfileResponseDto updateProfile(String email, UpdateProfileRequestDto request);
}

File: server/src/main/kotlin/com/smartexpense/api/application/port/in/ProfileUseCase.kt
package com.smartexpense.api.application.port.`in`

import com.smartexpense.api.application.dto.request.UpdateProfileRequestDto
import com.smartexpense.api.application.dto.response.UserProfileResponseDto

interface ProfileUseCase {
    fun getProfile(email: String): UserProfileResponseDto
    fun updateProfile(email: String, request: UpdateProfileRequestDto): UserProfileResponseDto
}
